backend/internal/core/services/deployment_service.go

```diff
@@ -244,7 +277,11 @@ func (svc *DeploymentService) AsyncRemoveNode(config statemanager.ClientConfig,
 		"node_name": nodeName,
 	}
 
-	if err = svc.runAsyncWithQueue(config.UserID, wf); err != nil {
+	if err = persistence.SetStateUserID(&wf, config.UserID); err != nil {
+		return "", "", err
+	}
+
+	if err = svc.runWithQueue(queueName, &wf); err != nil {
```

Comment:
I think all of the deployment service methods are doing the same code fundamentally with just a change of the workflow name. can we move all of that into a private generic method?